patches/server/0268-Prevent-Mob-AI-Rules-from-Loading-Chunks.patch

```diff
@@ -4,6 +4,18 @@ Date: Mon, 10 Sep 2018 23:56:36 -0400
 Subject: [PATCH] Prevent Mob AI Rules from Loading Chunks
 
 
+diff --git a/src/main/java/net/minecraft/world/entity/ai/goal/MoveToBlockGoal.java b/src/main/java/net/minecraft/world/entity/ai/goal/MoveToBlockGoal.java
+index 70a51ba19fb34f652858b18f24554261787d97e2..1a1f885ab5a181d5552e3c2ec2231086070467e1 100644
+--- a/src/main/java/net/minecraft/world/entity/ai/goal/MoveToBlockGoal.java
++++ b/src/main/java/net/minecraft/world/entity/ai/goal/MoveToBlockGoal.java
+@@ -107,6 +107,7 @@ public abstract class MoveToBlockGoal extends Goal {
+                 for(int m = 0; m <= l; m = m > 0 ? -m : 1 - m) {
+                     for(int n = m < l && m > -l ? l : 0; n <= l; n = n > 0 ? -n : 1 - n) {
+                         mutableBlockPos.setWithOffset(blockPos, m, k - 1, n);
++                        if (!this.mob.level.hasChunkAt(blockPos)) continue; // Paper - it's an invalid target if it's not loaded
```

Comment:
`blockPos` is the position of the mob, and `mutableBlockPos` is the potential target. Shouldn't `mutableBlockPos` be the one getting checked here?

Reply:
*I only approved this so others can look at it and merge it without waiting for core*
{Welp, that's my excuse}

But, yea, that looks to be checking the wrong thing, the mutable one is the one that is actually piped into logic which would load stuff

Reply:
It's funny I've had this patch written for weeks and as soon as I recreate it I forget everything. Thank you for pointing that out, pushed fix.